lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -411,43 +411,53 @@ protected function _getCreateTableSQL($table, array $columns, array $options = a
     }
 
     /**
+     * Returns the SQL for a list of indexes in the database.
+     *
+     * @param string $database
+     * @param string $table
+     *
+     * @return string
+     */
+    public function getListIndexesSQL(?string $database, ?string $table): string
+    {
+        if ($database !== NULL && $database !== '/') {
+            $databaseIdentifier = $this->normalizeIdentifier($database);
+            $quotedDatabaseIdentifier = $this->quoteStringLiteral($databaseIdentifier->getName());
+        }
+        else {
+            $quotedDatabaseIdentifier = "(SELECT SYS_CONTEXT('userenv', 'current_schema') FROM DUAL)";
```

Comment:
The point here is that we do not have a connection to query against in the Platform object.

Reply:
Then we should use `ALL_*` objects for the non-current schema and `USER_*` ones for the current one.

> Because in the new concept we use the `ALL_*` views that provide information about Oracle DBMS objects, instead of the slower `USER_*` views that do a lot of more checking about user being authorised to access an object (checked with EXPLAIN PLAN).

Is using `USER_*` slower than using a sub-query on each row of `ALL_*`?